plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -260,6 +266,7 @@ public LocationEngine getLocationEngine() {
    */
   @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
   public void onStop() {
+    StaleStateRunnable.getInstance().removeOnLocationStaleListener(null);
     StaleStateRunnable.getInstance().onStop();
```

Comment:
Calling getInstance twice while one time is sufficient, either assign result/reuse or refactor StateStateRunnable to encapsulate behaviour into one method. If `removeOnLocationStaleListener ` is only used in this code path the latter suggestion would make most sense.